### PR TITLE
Update Functionality Test 4 for Appium 2

### PR DESCRIPTION
**Pull request by geriux**
## Related PRs
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6276
- https://github.com/WordPress/gutenberg/pull/55166
- Part of https://github.com/WordPress/gutenberg/pull/55420

## What?
This PR updates the Functionality Test 4 tests to WebdriverIO and Appium 2.

## Why?
To continue with the migration effort to support this new version and new driver.

## How?
- Replace the usage of `sleep` in favor of `pause`
- Updates the usage of one test to pass the TextInput element to `takeScreenshotByElement`
- Snapshots were updated

## Testing Instructions
Local tests should pass by running the following command for both platforms:

**Android**
```
 TEST_RN_PLATFORM=android npm run device-tests:local gutenberg-editor-functionality-test-4-visual.test
```

**iOS**
```
 TEST_RN_PLATFORM=ios npm run device-tests:local gutenberg-editor-functionality-test-4-visual.test
```

I've also confirmed it runs on SauceLabs by triggering the test from a local command against the Appium 2 build.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: __device-tests__/image-snapshots/gutenberg-editor-functionality-test-4-visual-test-js-gutenberg-editor-test-suite-4-buttons-block-should-allow-edit-text-styles-1-android.png
Comment (dcalhoun) [inline __device-tests__/image-snapshots/gutenberg-editor-functionality-test-4-visual-test-js-gutenberg-editor-test-suite-4-buttons-block-should-allow-edit-text-styles-1-android.png:1]: It appears the button elements shifted slightly here and in the dark mode snapshot. Do you know why that is the case?

**Review by dcalhoun — APPROVED** [review]
Verified the test plan succeeded. 🚀

File: __device-tests__/image-snapshots/gutenberg-editor-functionality-test-4-visual-test-js-gutenberg-editor-test-suite-4-buttons-block-should-allow-edit-text-styles-1-android.png
Comment (geriux) [reply __device-tests__/image-snapshots/gutenberg-editor-functionality-test-4-visual-test-js-gutenberg-editor-test-suite-4-buttons-block-should-allow-edit-text-styles-1-android.png:1]: It looks like before the migration we were adding a space at the beginning of the text, if you see the previous screenshot, the text is not aligned. We've changed quite a few utils so it seems we had a bug somewhere 🤔

**Comment by geriux** [comment]
I'm going to merge this PR since the test failure is not related to this change. I'll create a different PR to fix that flaky test.

**Pull request merged**